Use JDK 11 in default images

MarkEWaite opened the pull request:
## Use JDK 11 in default images

As part of the 2.303.1 switch from Java 8 to Java 11 in the base images, this pull request adds Java 11 targets for Alpine, CentOS7, and Debian slim to the already existing Debian and UBI Java 11 images.  Java 8 images will continue to be generated with the '-jdk8' suffix.

Also updates from Java 8u292 to Java 8u302 now that releases are available for amd64, aarch64, ppc64le, and s390x.

Also updates from Java 11.0.11 to Java 11.0.12 now that releases are available for amd64, aarch64, ppc64le, and s390x.

Updates the UBI version from 8.4-206 to 8.4-209

Uses Jenkins 2.303 as the base image so that my development work was assured of doing a full build.  That change can be reverted if it is a distraction.

Removes the redundant "-lts" in tags that already include a version number, since the version number is clearly an LTS version number.  For example, the tag for 2.303.1-lts-slim will be 2.303.1-slim.

Removes some tag duplication that resulted from the transition to Java 11 and from the removal of the "-lts" string from tags that contain a version number.

timja commented on docker-bake.hcl:
```diff
@@ -115,13 +131,26 @@ target "centos7_jdk8" {
+  tags = [
+    "${REGISTRY}/${JENKINS_REPO}:${JENKINS_VERSION}-centos7",
+    equal(LATEST_WEEKLY, "true") ? "${REGISTRY}/${JENKINS_REPO}:centos7" : "",
+    equal(LATEST_LTS, "true") ? "${REGISTRY}/${JENKINS_REPO}:lts-centos7" : "",
```
do we need a jdk11 tag?

timja commented on docker-bake.hcl:
```diff
@@ -193,13 +216,26 @@ target "debian_slim_jdk8" {
+  }
+  tags = [
+    "${REGISTRY}/${JENKINS_REPO}:${JENKINS_VERSION}-slim",
+    equal(LATEST_WEEKLY, "true") ? "${REGISTRY}/${JENKINS_REPO}:slim" : "",
```
do we need a slim-jdk11 tag?

MarkEWaite replied:
Good point.  I had not included the addition of -jdk11 suffix tags for the other images, but it makes sense.  Should there be an additional tag for jdk11 on each of the images that uses JDK 11?

MarkEWaite replied:
I've added the `-jdk11` suffix to images in35b8d66aec21601a5a6b0f1aeb74f949549a7e32 .  If that's not what was intended, I can delete that commit.

timja replied:
that's what i meant

timja commented on docker-bake.hcl:
```diff
@@ -178,7 +202,6 @@ target "debian_jdk11" {
     "${REGISTRY}/${JENKINS_REPO}:${JENKINS_VERSION}-jdk11",
```
aren't we missing the `latest` tag here?

MarkEWaite replied:
Thanks for catching that.  Fixed in e08ebe035a7d1ec4f80c7b03b4aaa2815b61659f

timja reviewed (commented):
i think the latest tag is missing and is required, unless I'm missing somethign

slide commented:
Are you not making the same changes for the Windows side? It would be good to keep them in sync.

MarkEWaite commented:
> i think the latest tag is missing and is required, unless I'm missing somethign

Thanks very much for detecting that mistake!  I had dropped the `latest`, `lts`, and `2.303.1` tags.  They should have been moved to the Debian JDK 11 and were not.  Fixed in e08ebe035a7d1ec4f80c7b03b4aaa2815b61659f

MarkEWaite commented:
> Are you not making the same changes for the Windows side? It would be good to keep them in sync.

If I can understand the Windows Powershell script, I'll try to keep it in sync.  I'm not Powershell skilled, so I worry that I'll make some glaring mistake

timja commented:
The windows images aren’t publishing any fixed tag afaik just updating the same one all the time

timja commented on docker-bake.hcl:
```diff
@@ -2,11 +2,14 @@ group "linux" {
+    "alpine_jdk11",
     "centos7_jdk8",
+    "centos7_jdk11",
     "centos8_jdk8",
```
aren't we removing this?

MarkEWaite replied:
I thought we were retaining CentOS 7 because it is "not dead yet", while we are dropping CentOS 8 because it is clearly dead (no Docker image for CentOS 8.4).

timja replied:
This is centos8?

MarkEWaite replied:
My mistake in misinterpreting the line your question was addressing.  I think you were asking about the `centos-jdk8` entry and I mistakenly responded as though you were asking about `centos7_jdk11`.

I looked at the JEP draft and it says that we're stopping the publishing of CentOS 8 images.  I've included that in 7ef213bfc78978364c48e7000bd2f55321c2e7eb .  Thanks for catching it!

timja replied:
I've pushed https://github.com/jenkinsci/docker/pull/1178/commits/e2ea833308ff1b749f63b9a5ab1894fbcc25b4d0 to fully address this
